moto/route53/models.py

```diff
@@ -480,14 +480,28 @@ def create_hosted_zone(
             comment=comment,
             delegation_set=delegation_set,
         )
+        # For each public hosted zone that you create, Amazon Route 53 automatically creates a name server (NS) record
+        # and a start of authority (SOA) record.
+        # https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/SOA-NSrecords.html
+        soa_record_set = {
+            "Name": f"{name}.",
```

Comment:
```suggestion
            "Name": f"{name}" + ("" if name.endswith(".") else "."),
```
The name sometimes already ends with a dot, causing the record set to look like `<Name>hrkd1lvc.test..</Name>`. 
That in turn makes it impossible to delete the record, because TF will only look for `hrkd1lvc.test.`